tests/qnodes/test_qnode_reversible.py

```diff
@@ -0,0 +1,849 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the PennyLane :class:`~.ReversibleQNode` class.
+"""
+import pytest
+import numpy as np
+
+import pennylane as qml
+from pennylane._device import Device
+from pennylane.operation import CVObservable
+from pennylane.qnodes.base import QuantumFunctionError
+from pennylane.qnodes.rev import ReversibleQNode
+
+
+thetas = np.linspace(-2 * np.pi, 2 * np.pi, 8)
+
+
+@pytest.fixture(scope="function")
+def operable_mock_device_2_wires(monkeypatch):
+    """A mock instance of the abstract Device class that can support qfuncs."""
+
+    dev = Device
+    with monkeypatch.context() as m:
+        m.setattr(dev, "__abstractmethods__", frozenset())
+        m.setattr(dev, "_capabilities", {"model": "qubit"})
+        m.setattr(
+            dev, "operations", ["BasisState", "RX", "RY", "CNOT", "Rot", "PhaseShift"]
+        )
+        m.setattr(dev, "observables", ["PauliX", "PauliY", "PauliZ"])
+        m.setattr(dev, "reset", lambda self: None)
+        m.setattr(dev, "apply", lambda self, x, y, z: None)
+        m.setattr(dev, "expval", lambda self, x, y, z: 1)
+        yield Device(wires=2)
+
+
+class TestBestMethod:
+    """Test different flows of _best_method"""
+
+    def test_no_following_observable(self, operable_mock_device_2_wires):
+        """Test that the gradient is 0 if no observables succeed"""
+
+        def circuit(x):
+            qml.RX(x, wires=[1])
+            return qml.expval(qml.PauliZ(0))
+
+        q = ReversibleQNode(circuit, operable_mock_device_2_wires)
+        q._construct([1.0], {})
+        assert q.par_to_grad_method == {0: "0"}
+
+    def test_param_unused(self, operable_mock_device_2_wires):
+        """Test that the gradient is 0 of an unused parameter"""
+
+        def circuit(x, y):
+            qml.RX(x, wires=[0])
+            return qml.expval(qml.PauliZ(0))
+
+        q = ReversibleQNode(circuit, operable_mock_device_2_wires)
+        q._construct([1.0, 1.0], {})
+        assert q.par_to_grad_method == {0: "A", 1: "0"}
+
+    def test_not_differentiable(self, operable_mock_device_2_wires):
+        """Test that an operation with grad_method=None is marked as
+        non-differentiable"""
+
+        def circuit(x):
+            qml.BasisState(x, wires=[1])
+            return qml.expval(qml.PauliZ(0))
+
+        q = ReversibleQNode(circuit, operable_mock_device_2_wires)
+        q._construct([np.array([1.0])], {})
+        assert q.par_to_grad_method == {0: None}
```

Comment:
I don't think these tests are needed. They are testing the `QubitQNode.best_method()` method that is just inherited here, and is already being tested in the `QubitQNode` tests.

Better to leave tests like this in `test_qnodes_qubit` --- that way, changes to the QubitQNode logic require updating just the corresponding QubitQNode unit tests.